metrics/web/test/instant_config/presentation/state/instant_config_notifier_test.dart

```diff
@@ -0,0 +1,201 @@
+      () {
+        expect(
+          () => InstantConfigNotifier(_fetchInstantConfigUseCase),
+          returnsNormally,
+        );
+      },
+    );
+
+    test(
+      "creates an instance with the given fetch instant config use case",
+      () {
+        expect(
+          () => InstantConfigNotifier(_fetchInstantConfigUseCase),
+          returnsNormally,
+        );
+      },
+    );
+
+    test(
+      ".setDefaults() throws an AssertionError if the given is login form enabled is null",
+      () {
+        expect(
+          () => notifier.setDefaults(
+            isLoginFormEnabled: null,
+            isFpsMonitorEnabled: isFpsMonitorEnabled,
+            isRendererDisplayEnabled: isRendererDisplayEnabled,
+          ),
+          MatcherUtil.throwsAssertionError,
+        );
+      },
+    );
+
+    test(
+      ".setDefaults() throws an AssertionError if the given is FPS monitor enabled is null",
+      () {
+        expect(
+          () => notifier.setDefaults(
+            isLoginFormEnabled: isLoginFormEnabled,
+            isFpsMonitorEnabled: null,
+            isRendererDisplayEnabled: isRendererDisplayEnabled,
+          ),
+          MatcherUtil.throwsAssertionError,
+        );
+      },
+    );
+
+    test(
+      ".setDefaults() throws an AssertionError if the given is renderer display enabled is null",
+      () {
+        expect(
+          () => notifier.setDefaults(
+            isLoginFormEnabled: isLoginFormEnabled,
+            isFpsMonitorEnabled: isFpsMonitorEnabled,
+            isRendererDisplayEnabled: null,
+          ),
+          MatcherUtil.throwsAssertionError,
+        );
+      },
+    );
+
+    test(
+      ".initializeInstantConfig() calls the fetch instant config use case",
+      () {
+        when(_fetchInstantConfigUseCase(any)).thenAnswer(
+          (_) => Future.value(instantConfig),
+        );
+
+        notifier.initializeInstantConfig();
+
+        verify(_fetchInstantConfigUseCase(any)).called(1);
+      },
+    );
+
+    test(
+      ".initializeInstantConfig() uses the given default values when fetching the instant config",
+      () {
+        when(_fetchInstantConfigUseCase(any)).thenAnswer(
+          (_) => Future.value(instantConfig),
+        );
+
+        final instantConfigParam = InstantConfigParam(
+          isLoginFormEnabled: isLoginFormEnabled,
+          isFpsMonitorEnabled: isFpsMonitorEnabled,
+          isRendererDisplayEnabled: isRendererDisplayEnabled,
+        );
+
+        notifier.initializeInstantConfig();
+
+        verify(_fetchInstantConfigUseCase(instantConfigParam)).called(1);
+      },
+    );
+
+    test(
+      ".initializeInstantConfig() sets the login form instant config view model",
+      () async {
+        when(_fetchInstantConfigUseCase(any)).thenAnswer(
+          (_) => Future.value(instantConfig),
+        );
+
+        await notifier.initializeInstantConfig();
+
+        expect(notifier.loginFormInstantConfigViewModel, isNotNull);
```

Comment:
Maybe is makes sense to check if the `isEnabled` value equals to one we've fetched?